docs/source/tracking.rst

```diff
@@ -76,6 +76,104 @@ There are different kinds of remote tracking URIs:
+interfaces with them to record MLflow entities and artifacts:
+
+ * `MLflowClient` creates an instance of a `LocalArtifactRepository` (to store artifacts)
+ * `MLflowClient` creates an instance of a `FileStore` (to save MLflow entities)
+
+Scenario 2: MLflow on the localhost with SQLite
+-----------------------------------------------
+
+A slight variation from scenario 1, MLflow runs on a localhost or laptop but with backend store as an SQLAlchemy compatible
+database type: **SQLite**. Artifacts are stored under the local ``./mlruns`` directory, and MLflow entities are
+inserted in an SQLite database file ``mlruns.db``.
+
+.. figure:: _static/images/scenario_2.png
+
+As above, `MLflowClient` is responsible to create respective instances:
+
+ * `MLflowClient` creates an instance of a `LocalArtifactRepository` (to save artifacts)
+ * `MLflowClient` creates an instance of an `SQLAlchemyStore` (to save MLflow entities) and writes to SQLite file ``mlruns.db``)
+
+Scenario 3: MLflow on the localhost with Tracking Server
+--------------------------------------------------------
+
+Similar to scenario 1 but a tracking server is launched, listening for REST request calls at the default port 5000.
+As in scenario 1, MLflow uses the local `file://path/mlruns` as a backend store and artifact store. With a tracking
+server running, the `MLflowClient` will interact with tracking server via REST requests as shown in the diagram.
+
+.. figure:: _static/images/scenario_3.png
+
+To store all runs' MLflow entities, the `MLflowClient` interacts with the tracking server via a series of REST requests:
+
+ * **Part 1a and b**:
+
+  * `MLflowClient` creates an instance of a `RestStore`
+  * `MLflowClient` sends a REST API request to log MLflow entities
+  * Tracking Server creates an instance of a `FileStore` (to save MLflow entities) and writes directly
+    to the local `file://path/mlruns` directory.
+
+For the artifacts, the `MLflowClient` interacts with the tracking server via a REST request:
+
+ * **Part 2a, b, and c**:
+
+   * `MLflowClient` uses `RestStore` to send a REST request to fetch the artifact store URI location
+   * Tracking Server responds with an artifact store URI location
+   * `MLflowClient` creates an instance of a `LocalArtifactRepository` (to save artifacts at the local URI location)
+
+
+Scenario 4: MLflow with remote Tracking Server, backend and artifact stores
+---------------------------------------------------------------------------
+
+This is a common MLflow setup where the tracking server is remote, to which clients
+send REST requests; a remote SQLAlchemy compatible `database-backend store <https://mlflow.org/docs/latest/tracking.html#backend-stores>`_, to which the tracking
+server records MLflow entities; and a remote, supported and scalable `artifact store <https://mlflow.org/docs/latest/tracking.html#artifact-stores>`_, to which the clients
+upload artifacts.
```

Comment:
I think it would help to describe the general capability of MLflow to work in distributed environments and indicate that this is an example.

E.g.

```suggestion
MLflow also supports distributed architectures, where the Tracking Server, backend store, and artifact store reside on remote hosts. This example scenario depicts an architecture with a remote MLflow Tracking Server, a Postgres database for backend entity storage, and an S3 bucket for artifact storage.
```

Reply:
You don't like my parallel structure of using infinitive phrases ... :-) I Will update it with this distributed scenario.

Reply:
@dbczumar Thanks for detailed feedback. I have incorporated all nits and suggestions, including much crispier diagrams—the ppt did the trick.